src/qt/receivecoinsdialog.cpp

```diff
@@ -287,6 +290,19 @@ void ReceiveCoinsDialog::copyURI()
     GUIUtil::setClipboard(uri);
 }
 
+// context menu action: copy address
+void ReceiveCoinsDialog::copyAddress()
+{
+    const QModelIndex sel = selectedRow();
+    if (!sel.isValid()) {
+        return;
+    }
+
+    const RecentRequestsTableModel * const submodel = model->getRecentRequestsTableModel();
```

Comment:
style nit (see [clang-format-diff.py](https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy)):
```suggestion
    const RecentRequestsTableModel* const submodel = model->getRecentRequestsTableModel();
```

---

btw, why `submodel` variable is required at all? It does not improve readability, at least for me :)
Maybe just
```
    const QString address = model->getRecentRequestsTableModel()->entry(sel.row()).recipient.address;
```

This line could be [shorter](https://github.com/bitcoin/bitcoin/pull/20986) then 100 characters, if s/`sel`/`r`

Feel free to ignore the second part of this comment.

Reply:
addressed the first half of your comment in e348d7e